test/dialect/AIEVec/precanonicalization.mlir

```diff
@@ -66,3 +66,44 @@ func.func @rank_zero_transfer_read(%m : memref<i16>) -> vector<16xi16> {
     // CHECK: return %[[S]] : vector<16xi16>
     return %v : vector<16xi16>
 }
+
+// -----
+
+// CHECK-LABEL: func.func @extsi_hoisting_through_extract_n_bcast(
+// CHECK-SAME: %[[VEC:.*]]: vector<16xi8>
+func.func @extsi_hoisting_through_extract_n_bcast(%v : vector<16xi8>)
+                                                            -> vector<32xi32> {
+// CHECK: %[[EXV:.*]] = arith.extsi %[[VEC]] : vector<16xi8> to vector<16xi32>
+// CHECK: %[[EXS:.*]] = vector.extract %[[EXV]][7] : vector<16xi32>
+// CHECK: %[[BCAST:.*]] = vector.broadcast %[[EXS]] : i32 to vector<32xi32>
+// CHECK: return %[[BCAST]] : vector<32xi32>
+  %si8 = vector.extract %v[7] : vector<16xi8>
+  %vi8 = vector.broadcast %si8 : i8 to vector<32xi8>
+  %vi32 = arith.extsi %vi8 : vector<32xi8> to vector<32xi32>
```

Comment:
Do you have the real user test cases which generate this extsi op pattern?

Reply:
I do, but it breaks for other reasons. Before it runs end-to-end, I need to fix the issue with Vector2AIEVec letting arithmetic vector operations through unconverted, because, predictably, the legalization pattern fails for this use case. Once that's solved, and together with this, it should work. I'll add the new test case with that (longer) patch.